domains/measurements/Measurements.API/API/Measurements/gRPC/V1/Services/MeasurementsParser.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using API.Measurements.Helpers;
+using Measurements.V1;
+using Metertimeseries.V1;
+
+namespace API.Measurements.gRPC.V1.Services;
+
+public class MeasurementsParser
+{
+    public List<Measurement> ParseMeasurements(GetMeasurementsRequest request, MeterTimeSeriesResponse dhResponse)
+    {
+        var meteringPoints = dhResponse.GetMeterTimeSeriesResult.MeterTimeSeriesMeteringPoint;
+        var expectedMeasurementCount = (int)(request.DateTo - request.DateFrom) / 3600;
+        var result = new List<Measurement>(Math.Max(expectedMeasurementCount, 0));
+
+        foreach (var meteringPoint in meteringPoints)
+        {
+            // TODO: Assumes a single metering point in each request
```

Comment:
Valid assumption - we query pr gsrn

Reply:
Remove comment and added filter for correct GSRN.